internal/kafka/command_topic_describe_onprem.go

```diff
@@ -55,12 +68,63 @@ func DescribeTopic(cmd *cobra.Command, restClient *kafkarestv3.APIClient, restCo
 		return kafkarest.NewError(restClient.GetConfig().BasePath, err, resp)
 	}
 
-	table := output.NewTable(cmd)
-	table.Add(&topicOut{
+	topicDescribe := &topicOut{
 		Name:              topic.TopicName,
 		IsInternal:        topic.IsInternal,
 		ReplicationFactor: topic.ReplicationFactor,
 		PartitionCount:    topic.PartitionsCount,
-	})
-	return table.Print()
+	}
+
+	partitions, resp, err := restClient.PartitionV3Api.ListKafkaPartitions(restContext, clusterId, topicName)
+	if err != nil {
+		return kafkarest.NewError(restClient.GetConfig().BasePath, err, resp)
+	} else if partitions.Data == nil {
+		return errors.NewErrorWithSuggestions(errors.InternalServerErrorMsg, errors.InternalServerErrorSuggestions)
+	}
+
+	partitionList := make([]*PartitionData, len(partitions.Data))
+
+	for i, partition := range partitions.Data {
+		replicas, resp, err := restClient.ReplicaApi.ClustersClusterIdTopicsTopicNamePartitionsPartitionIdReplicasGet(restContext, clusterId, topicName, partition.PartitionId)
+		if err != nil {
+			return kafkarest.NewError(restClient.GetConfig().BasePath, err, resp)
+		} else if replicas.Data == nil {
+			return errors.NewErrorWithSuggestions(errors.InternalServerErrorMsg, errors.InternalServerErrorSuggestions)
+		}
+		partitionList[i] = &PartitionData{
+			PartitionId: partition.PartitionId,
+			Replicas:    make([]int32, len(replicas.Data)),
+			ISR:         make([]int32, 0, len(replicas.Data)),
+		}
+		for j, replica := range replicas.Data {
+			if replica.IsLeader {
+				partitionList[i].Leader = replica.BrokerId
+			}
+			partitionList[i].Replicas[j] = replica.BrokerId
+			if replica.IsInSync {
+				partitionList[i].ISR = append(partitionList[i].ISR, replica.BrokerId)
+			}
+		}
+	}
+
+	if output.GetFormat(cmd).IsSerialized() {
+		return output.SerializedOutput(cmd, &describeOutOnPrem{
+			topicOut:   *topicDescribe,
+			Partitions: partitionList,
+		})
+	}
+
+	table := output.NewTable(cmd)
+	table.Add(topicDescribe)
+	if err := table.Print(); err != nil {
+		return err
+	}
+
+	output.Println(false, "")
+
+	list := output.NewList(cmd)
+	for _, partition := range partitionList {
+		list.Add(partition)
+	}
+	return list.Print()
```

Comment:
I'm thinking no, because there's already `confluent kafka partition list` and `confluent kafka replica list` commands for on-prem which give us naming conflicts if we try to split this out.

And it would be a bit awkward to make a `confluent kafka topic [partition| replica] list` when the information here is essentially the same as the output of `confluent kafka replica list`, with the only difference being this is available even for clusters where the other endpoint in `confluent kafka replica list`.

Reply:
> with the only difference being this is available even for clusters where the other endpoint in confluent kafka replica list

What do you mean?

Reply:
`confluent kafka replica list` uses the Replica Status API that is only available for brokers deployed with Confluent Server, which is only available with the enterprise license. The current `confluent kafka topic describe` uses the Replica API that's available for any broker, so people using the community license can still get partition replica info from it.

Also, I should clarify: when I said that the information was essentially the same, that was misleading and what I really meant was that the Replica API output is basically a subset of the Replica Status API. But the latter returns more info, so we can't just replace it with the Replica API either.

Reply:
Based on these:
- https://docs.confluent.io/platform/current/kafka-rest/index.html#api-availability
- https://docs.confluent.io/platform/current/installation/license.html#license-types

Reply:
Thanks, I remember now. A few options:
1. Query the replica status API, and if that doesn't work (the user is not enterprise) query the replica API
2. Query another API (if it exists?) to determine if the user is enterprise or not

What information would we be losing if we only queried the replica API?

Reply:
> What information would we be losing if we only queried the replica API?

Quite a lot: `is_observer`, `is_isr_eligible`, `is_caught_up`, `log_start_offset`, `log_end_offset`, `last_caught_up_time_ms`, `last_fetch_time_ms`, and `link_name`.

If we went with option 1, I guess we could just leave the other fields blank.

Reply:
Option 1 sounds good, and we can hide those columns so it doesn't look like any important information is missing.